libusb/hid.c

```diff
@@ -245,19 +257,23 @@ static int get_usage(uint8_t *report_descriptor, size_t size,
 		}
 
 		if (key_cmd == 0x4) {
-			*usage_page  = get_bytes(report_descriptor, size, data_len, i);
+			if (usage_page)
```

Comment:
I'd rather have all of these checks at the beginning of the function
`get_usage` is designed for internal usage only, so all these checks are really shouldn't be `required` (if it where a C++ - it would have been references, which do not require checks at all), but I don't mind having those as saniti checks, for as long as it doesn't affect the actual implementation

Reply:
I wasn't sure if there was a use case where you might not care about these outputs.
I can just early out at the beginning of these are supposed to be valid